models/coded.go

```diff
@@ -86,16 +86,18 @@ func computeIntersection(a []string, b []string) []string {
 	return intersect
 }
 
-func (c *Coded) PreferredCode(preferredCodeSets []string, codeSetRequired bool, valueSetPreferred bool, vsMap ValueSetMap) Concept {
+func (c *Coded) PreferredCode(preferredCodeSets []string, codeSetRequired bool, valueSetPreferred bool, vsMap ValueSetMap, mdcOid string) Concept {
 	if len(c.Codes) == 0 {
 		return Concept{}
 	}
+	mdcValueSet := vsMap[mdcOid]
 	if valueSetPreferred {
 		for codeSystem := range c.Codes {
 			for _, code := range c.Codes[codeSystem] {
 				for _, vsOid := range preferredCodeSets {
-					valueSet := vsMap[vsOid]
-					if len(valueSet) == 0 || codeSetContainsCode(valueSet, CodedConcept{CodeSystem: codeSystem, Code: code}) {
```

Comment:
if (len(prefValueSet) == 0 || codeSetContainsCode(prefValueSet, CodedConcept{CodeSystem: codeSystem, Code: code})) && codeSetContainsCode(mdcValueSet, CodedConcept{CodeSystem: codeSystem, Code: code}) {

The Code ALWAYS needs to be in the mdcValueSet.  The original OR placement does not enforce that.